lib/facter/resolvers/windows/timezone.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+module Facter
+  module Resolvers
+    module Windows
+      class Timezone < BaseResolver
+        init_resolver
+
+        class << self
+          private
+
+          def post_resolve(fact_name, _options)
+            @fact_list.fetch(fact_name) { determine_timezone }
+          end
+
+          def determine_timezone
+            timezone = Time.now.zone
+            @fact_list[:timezone] = timezone.force_encoding("CP#{codepage}").encode('UTF-8')
```

Comment:
```suggestion
            @fact_list[:timezone] = timezone.force_encoding("CP#{codepage}").encode('UTF-8', invalid: :replace)
```

maybe we can do this so we don't end up (even by accident) in a case where the bytes aren't valid UTF-8, and we end up breaking all Facter